nextstep-di/src/main/java/nextstep/di/factory/BeanFactory.java

```diff
@@ -70,11 +70,11 @@ private Object putParameterizedObject(Class<?> preInstanticateBean, Constructor<
 
     public Map<Class<?>, Object> getAnnotatedWith(Class<? extends Annotation> annotation) {
         Map<Class<?>, Object> annotatedClass = Maps.newHashMap();
-        for (Class<?> clazz : beans.keySet()) {
-            if (clazz.isAnnotationPresent(annotation)) {
-                annotatedClass.put(clazz, beans.get(clazz));
-            }
-        }
+
+        beans.keySet().stream()
+                .filter(clazz -> clazz.isAnnotationPresent(annotation))
+                .forEach(clazz -> annotatedClass.put(clazz, beans.get(clazz)));
```

Comment:
forEach 대신 Collectors.toMap을 사용해보면 어떨까요?
```suggestion
                .collect(Collectors.toMap(clazz -> clazz, beans::get));
```

stream API를 사용하실 때 참고하시면 좋을 아티클과 책의 요약글 함께 첨부드려요.
- [[이펙티브 자바3판] 7장 람다와 스트림](https://sjh836.tistory.com/173)
- [Java8 Stream은 loop가 아니다.](https://www.popit.kr/java8-stream%EC%9D%80-loop%EA%B0%80-%EC%95%84%EB%8B%88%EB%8B%A4/)

Reply:
Collectors 를 사용하니 더 깔끔한 코드가 되었습니다. stream 의 for each를 조심히 사용해야한다는 것을 알았습니다. 좋은 요약글 추천해주셔서 감사합니다 :)